src/lib.rs

```diff
@@ -63,156 +69,159 @@ pub fn detect_indent(string: &str) -> Indent {
     }
 
     let mut spaces = 0;
-    let mut tabs   = 0;
+    let mut tabs = 0;
     let mut indents: HashMap<isize, Usage> = HashMap::new();
 
     let mut prev = 0;
-    let mut current : Option<isize> = None;
+    let mut current: Option<isize> = None;
     let mut key;
 
     for line in string.lines() {
-        if line.is_empty() { continue; }
+        if line.is_empty() {
+            continue;
+        }
         let mut indent = 0;
 
         match INDENT_REGEX.captures(line) {
             Some(captures) => {
-                captures.get(0).map( |capture| {
+                if let Some(capture) = captures.get(0) {
```

Comment:
I'm open to this change, but mind sharing why I should do this over the former?  I want to incorporate the rationale into my model

Reply:
This was actually a clippy lint suggestion. I'm not positive of how it impacts the final emitted output from LLVM etc., but I wouldn't be surprised if it gets to do something simpler by using `if let` rather than using the function with callback. LLVM and the various lowering passes might be smart enough to skip the function allocation, but probably only for specific scenarios; and there's *definitely* never an allocation with the `if let`, so I suspect that's the reason for the lint.